### PR TITLE
Allow disabling Reaper auth

### LINKED ISSUE #1160: Allow disabling Reaper UI auth
Disabling Reaper's auth can theoretically be done by adding an empty name for `Reaper.spec.UiUserSecretRef`. In k8ssandra-operator though, we create a secret with credentials that's then added as a ref.
We need to introduce a way of disabling the auth for Reaper's UI to accommodate some use cases.

### PR DESCRIPTION
**Pull request by Miles-Garnsey**
Allow a user to disable Reaper auth by specifically setting the UiUserSecretRef.Name to "".

Fixes #1160

### CODE REVIEW

**Comment by adejanovski** [comment]
@Miles-Garnsey, FYI I moved this PR to draft since CI is failing and there seems to be a little more work before it can be reviewed.

**Comment by Miles-Garnsey** [comment]
> @Miles-Garnsey, FYI I moved this PR to draft since CI is failing and there seems to be a little more work before it can be reviewed.

I'm hoping these are just flakes actually. I've seen flakes on all the failing tests in the last few weeks.

**Comment by Miles-Garnsey** [comment]
OK, got tests passing locally, still some manual testing to do if CI goes green for us.

**Comment by Miles-Garnsey** [comment]
Still testing this, there is some weirdness where the secrets don't appear to be deleted after the cluster is deleted. That's confusing things a little in the case where auth is toggled off and on (the only test still failing is around this case).

**Comment by Miles-Garnsey** [comment]
I've manually tested this and it all appears to work now. Ready for review subject to the CI run passing.

File: apis/reaper/v1alpha1/reaper_types.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ type ReaperTemplate struct {
 
 	// Defines the secret which contains the username and password for the Reaper UI and REST API authentication.
 	// +optional
-	UiUserSecretRef corev1.LocalObjectReference `json:"uiUserSecretRef,omitempty"`
+	UiUserSecretRef *corev1.LocalObjectReference `json:"uiUserSecretRef,omitempty"`
```
Comment (adejanovski) [inline apis/reaper/v1alpha1/reaper_types.go:59]: **Todo:** Could you add a comment here explaining how to disable auth? That'll end up in the CRD docs.

**Review by adejanovski — APPROVED** [review]
**Todo:** Add a changelog entry.

Just two small things to do and you're good for a merge.

File: apis/reaper/v1alpha1/reaper_types.go (hunk 1 above)
Comment (Miles-Garnsey) [reply apis/reaper/v1alpha1/reaper_types.go:59]: Sure.

File: controllers/k8ssandra/secrets.go [hunk 2]
```diff
@@ -66,22 +66,26 @@ func (r *K8ssandraClusterReconciler) reconcileReaperSecrets(ctx context.Context,
...
 				return result.Error(err)
 			}
-			if err := secret.ReconcileSecret(ctx, r.Client, uiUserSecretRef.Name, kcKey); err != nil {
-				logger.Error(err, "Failed to reconcile Reaper UI secret", "ReaperUiUserSecretRef", uiUserSecretRef)
-				return result.Error(err)
+			if kc.Spec.Reaper.UiUserSecretRef == nil || kc.Spec.Reaper.UiUserSecretRef.Name != "" {
+				if err := secret.ReconcileSecret(ctx, r.Client, uiUserSecretRef.Name, kcKey); err != nil {
+					logger.Error(err, "Failed to reconcile Reaper UI secret", "ReaperUiUserSecretRef", uiUserSecretRef)
+					return result.Error(err)
+				}
```
Comment (c3-clement) [inline controllers/k8ssandra/secrets.go:88]: Hello @Miles-Garnsey @adejanovski ,

Isn't this change backward incompatible?

I just opened a GitHub issue https://github.com/k8ssandra/k8ssandra-operator/issues/1269

**Pull request merged**